src/borg/repository.py

```diff
@@ -1475,6 +1476,20 @@ def delete_segment(self, segment):
         except FileNotFoundError:
             pass
 
+    def clear_empty_dirs(self):
+        "Delete empty data dirs, i.e those with no segment files"
+        data_dir = os.path.join(self.path, 'data')
+        segment_dirs = [f.path for f in os.scandir(data_dir) if f.is_dir()]
+        for segment_dir in segment_dirs:
+            if len([f for f in os.scandir(segment_dir)]) == 0:
+                try:
+                    os.rmdir(segment_dir)
+                    sync_dir(data_dir)
+                except FileNotFoundError:
+                    logger.debug("While trying to clear empty segments dir, it seems it already vanished")
+                except OSError:
+                    logger.debug('While trying to clear empty segments dir, it seems it is not empty after all')
+
```

Comment:
and even more simple would be to exploit the document behaviour of `os.rmdir`:

https://docs.python.org/3/library/os.html#os.rmdir

- it only removes empty dirs (exactly what we want)
- raises an OSError if directory not found or not empty

we don't need to debug log there, guess this is a valid case to just silently ignore these (expected) error cases.
but, we want a comment in the exception handler about all this.

Reply:
nothing is really so important that we need to sync_dir all the time, guess once at the end and we should be good.

Reply:
Ignoring the `OSError` as it is expected case. And keeping debug info for `FileNotFoundError` as it is not generally expected. Seems good?

sync_dir moved to end.